src/vivarium_public_health/dataset_manager/hdf.py

```diff
@@ -139,13 +137,10 @@ def _write_data_frame(path: str, entity_key: 'EntityKey', data: pd.DataFrame):
     if data.empty:
         raise ValueError("Cannot persist empty dataset")
 
-    # Even though these get called data_columns, it's more correct to think of them
-    # as the columns you can use to index into the raw data with. It's the subset of columns
-    # that you can filter by without reading in a whole dataset.
-    data_columns = DEFAULT_COLUMNS.intersection(data.columns)
-
+    # Our data is indexed (mostly -- demog dimensions isn't).
```

Comment:
hmm so here's something I just thought of: we will now no longer be able to filter demog dimensions by location without reading it all in. Is that going to cause problems for multi-loc artifacts?